deepchem/models/torch_models/molgan.py

```diff
@@ -0,0 +1,82 @@
+from typing import  List, Tuple #, Any, Optional,
+
+# import torch
+# from torch import nn
+# from torch.nn import functional as F
+# from deepchem.feat.molecule_featurizers.molgan_featurizer import GraphMatrix
+from deepchem.models import WGAN
+# from deepchem.models.layers import MolGANEncoderLayer
+
+
+class BasicMolGANModel(WGAN):
+    """
+    Model for de-novo generation of small molecules based on work of Nicola De Cao et al. [1]_.
+    It uses a GAN directly on graph data and a reinforcement learning objective to induce the network to generate molecules with certain chemical properties.
+    Utilizes WGAN infrastructure; uses adjacency matrix and node features as inputs.
+    Inputs need to be one-hot representation.
```

Comment:
Can you add a usage example here?

Reply:
I will add the example once i complete the porting of BasicMolGANModel class